react/style-guide.md

```diff
@@ -0,0 +1,948 @@
+  - **Filename**: Use PascalCase for filenames. E.g., `ReservationCard.js`.
+  - **Reference Naming**: Use PascalCase for React components and camelCase for their instances. eslint: [`react/jsx-pascal-case`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-pascal-case.md)
+
+    ```jsx
+    // bad
+    import reservationCard from './ReservationCard';
+
+    // good
+    import ReservationCard from './ReservationCard';
+
+    // bad
+    const ReservationItem = <ReservationCard />;
+
+    // good
+    const reservationItem = <ReservationCard />;
+    ```
+  - **Component Hierarchy**: 
+    - Component files should be inside folders that match the component's name.
+    - Use index.js as the filename of a container/smart component. Use `Container` as the suffix of the component's name.
+    - Use layout.js as the filename of a layout/dumb component.
+
+    
+    ```jsx
+    // MyComponent/index.js
+    import MyComponent from './layout'
+
+    class MyComponentContainer extends Component {
+      // Do smart stuff
+
+      render() {
+        return <MyComponent />
+      }
+    }
+
+    // MyComponent/layout.js
+    function MyComponent() {
+      return (
+        // Some JSX
+      )
+    }
+    ```
+  
+  - **Higher-order Component Naming**: Use a composite of the higher-order component's name and the passed-in component's name as the `displayName` on the generated component. For example, the higher-order component `withFoo()`, when passed a component `Bar` should produce a component with a `displayName` of `withFoo(Bar)`.
+
+    > Why? A component's `displayName` may be used by developer tools or in error messages, and having a value that clearly expresses this relationship helps people understand what is happening.
+
+    ```jsx
+    // bad
+    export default function withFoo(WrappedComponent) {
+      return function WithFoo(props) {
+        return <WrappedComponent {...props} foo />;
+      }
+    }
+
+    // good
+    export default function withFoo(WrappedComponent) {
+      function WithFoo(props) {
+        return <WrappedComponent {...props} foo />;
+      }
+
+      const wrappedComponentName = WrappedComponent.displayName
+        || WrappedComponent.name
+        || 'Component';
+
+      WithFoo.displayName = `withFoo(${wrappedComponentName})`;
+      return WithFoo;
+    }
+    ```
+
+  - **Props Naming**: Avoid using DOM component prop names for different purposes.
+
+    > Why? People expect props like `style` and `className` to mean one specific thing. Varying this API for a subset of your app makes the code less readable and less maintainable, and may cause bugs.
+
+    ```jsx
+    // bad
+    <MyComponent style="fancy" />
+
+    // bad
+    <MyComponent className="fancy" />
+
+    // good
+    <MyComponent variant="fancy" />
+    ```
+
+## Import
+
+  - Never use `import *`. Use an object to wrap the variables inside the file and then import that object.
```

Comment:
I think it's better to `import *` when you only need all of the elements exported and the few that you need when you only need some, than importing every item every time you need something from that file

Reply:
I've seen examples of it being used just to add more readability:

```
// routes.js
const DASHBOARD_PATH = '/dashboard';
//other file
import * as Routes from './routes.js'
const path = Routes.DASHBOARD_PATH;
```

And in this cases I like the readability improvement (even if we are importing more than we need), but if the variables in that path need that kind of readability when imported I prefer the use of the object:
```
// routes.js
const Routes = {
  DASHBOARD_PATH: '/dashboard';
}
//other file
import Routes from './routes.js'
const path = Routes.DASHBOARD_PATH;
```

You actually use objects all the time without needing 100% of its elements. Maybe what needs to be clarifies is that this is not a pattern to be used always, but something you'd do in specific occasions when you want more clarity

Reply:
I agree. Maybe ditch the word `Never` for `It's preferable to wrap the variables inside the file...`